modules/extension/wms/src/test/resources/org/geotools/ows/wms/xml/test/test-data/1.1.0CapabilitiesMissingLogoUrlFormat.xml

```diff
@@ -0,0 +1,110 @@
+        </HTTP>
+      </DCPType>
+    </GetMap>
+    <GetFeatureInfo>
+      <Format>text/plain</Format>
+      <Format>text/html</Format>
+      <Format>application/vnd.ogc.gml</Format>
+      <DCPType>
+        <HTTP>
+          <Get><OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://dev1.dmsolutions.ca/cgi-bin/mswms_gmap?"/></Get>
+          <Post><OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://dev1.dmsolutions.ca/cgi-bin/mswms_gmap?"/></Post>
+        </HTTP>
+      </DCPType>
+    </GetFeatureInfo>
+    <DescribeLayer>
+      <Format>text/xml</Format>
+      <DCPType>
+        <HTTP>
+          <Get><OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://dev1.dmsolutions.ca/cgi-bin/mswms_gmap?"/></Get>
+          <Post><OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://dev1.dmsolutions.ca/cgi-bin/mswms_gmap?"/></Post>
+        </HTTP>
+      </DCPType>
+    </DescribeLayer>
+    <GetLegendGraphic>
+      <Format>image/gif</Format>
+      <Format>image/png</Format>
+      <Format>image/jpeg</Format>
+      <Format>image/wbmp</Format>
+      <Format>text/html</Format>
+      <Format>application/x-pdf</Format>
+      <Format>image/tiff</Format>
+      <DCPType>
+        <HTTP>
+          <Get><OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://dev1.dmsolutions.ca/cgi-bin/mswms_gmap?"/></Get>
+          <Post><OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://dev1.dmsolutions.ca/cgi-bin/mswms_gmap?"/></Post>
+        </HTTP>
+      </DCPType>
+    </GetLegendGraphic>
+  </Request>
+  <Exception>
+    <Format>application/vnd.ogc.se_xml</Format>
+    <Format>application/vnd.ogc.se_inimage</Format>
+    <Format>application/vnd.ogc.se_blank</Format>
+  </Exception>
+  <VendorSpecificCapabilities />
+  <UserDefinedSymbolization SupportSLD="1" UserLayer="0" UserStyle="1" RemoteWFS="0"/>
+  <Layer>
+    <Name>DEMO</Name>
+    <Title>GMap WMS Demo Server</Title>
+    <Abstract>Abstract Test</Abstract>
+    <KeywordList>
+    	<Keyword>word1</Keyword>
+    	<Keyword>word2</Keyword>
+    </KeywordList>
+    <SRS>EPSG:42304 EPSG:42101 EPSG:4269 EPSG:4326</SRS>
+    <LatLonBoundingBox minx="-172.367" miny="35.6673" maxx="-11.5624" maxy="83.8293" />
+    <BoundingBox SRS="EPSG:42304"
+                minx="-2.2e+06" miny="-712631" maxx="3.0728e+06" maxy="3.84e+06" />
+    <Attribution>
+      <Title>test</Title>
+      <OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:type="simple" xlink:href="http://www.example.com"/>
+      <LogoURL height="100" width="100">
+            <!-- <Format>image/png</Format> --> <!-- Removed for GEOT-7611 test (ArrayIndexOutOfBoundsException when missing format) -->
+          <OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:type="simple" xlink:href="http://www.osgeo.org/sites/all/themes/osgeo/logo.png"/>
```

Comment:
So it is actually invalid, you are just trying to be forgiving?

Reply:
You are making a good point. I was concerned about the whole getcapabilities response parsing failing because of one malformed element for one layer. Even if we don't consider the fact that we don't use the format information later on, we could still consume the reste of the layers while having just that one failing.

The previous test written are using validation hints while it is disabled elsewhere : 
https://github.com/geotools/geotools/blob/7033c917fe2c3aedb02eb19a19f4034d4da868ff/modules/extension/wms/src/main/java/org/geotools/ows/wms/response/WMSGetCapabilitiesResponse.java#L51C1-L53C14

I thought that the fact that document validation is disabled meant that we are indeed trying to be forgiving.
The previous code works properly with validation enabled, because this ensures Format and OnlineResource elements presence and order. See prevous implementation : 
```
if (sameName(elems[1], value[i])) {
        url = (URL) value[1].getValue();
}
```
Because this validation is disabled outside tests, the code seems misleading to me. It is testing value[i] and then accessing value[1].

Reply:
I agree concerning LegendUrl. I just pushed the corresponding fix.